File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/PropertyGrid/Functionality/DemoObject.cs
using System;
using System.Data;
using System.Configuration;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
using System.ComponentModel;
using System.Drawing;

namespace CompanionKit.Controls.PropertyGrid.Functionality
{
    /// <summary>
    /// The enum to use to demostrate editing of a property in the PropertyGrid
    /// </summary>
    public enum DemoEnum
    {
        EnumValueA=1,
        EnumValueB=2,
        EnumValueC=4
    }

    /// <summary>
    /// A demo object to use to demostrate editing of a property 
    /// in the PropertyGrid.
    /// </summary>
    /// 
    [TypeConverter(typeof(DemoObjectConverter))]
    public class DemoObject
    {
        private int mintValueA = 1;
        private int mintValueB = 2;
        private int mintValueC = 3;
        private DemoEnum menmValueD = DemoEnum.EnumValueA;
        private Color menmColor = Color.LawnGreen;
        
        private string mstrValueA = "Initial A";
        private string mstrValueB = "Initial B";
        private string mstrValueC = "Initial C";

        [Category("Int Properties")]
        [Browsable(true)] // The default value
        [ReadOnly(true)]  // The default is read/write
        [Description("Integer ReadOnly property")]
        public int ValueA
        {
            get { return mintValueA; }
            set { mintValueA = value; }
        }
        
        [Category("Int Properties")]
        [Description("Integer Read/Write property")]
        public int ValueB
        {
            get { return mintValueB; }
            set { mintValueB = value; }
        }

        [Category("Advanced")]
        [Description("Integer Read/Write property in Advanced category")]
        public int ValueC
        {
            get { return mintValueC; }
            set { mintValueC = value; }
        }

        [Category("Strings")]
        [Description("String - 'A' Read/Write property in Strings category")]
        public string StringValueA
        {
            get { return mstrValueA; }
            set { mstrValueA = value; }
        }
        
        [Category("Strings")]
        [Description("String - 'B' Read/Write property in Strings category")]
        public string StringValueB
        {
            get { return mstrValueB; }
            set { mstrValueB = value; }
        }

        [Category("Advanced")]
        [Description("String - 'C' Read/Write property in Advanced category")]
        public string StringValueC
        {
            get { return mstrValueC; }
            set { mstrValueC = value; }
        }

        [Category("Enums")]
        [Description("The property with enum type")]
        public DemoEnum EnumProperty
        {
            get { return menmValueD; }
            set { menmValueD = value; }
        }


        [Category("Enums")]
        [Description("The property with enum type - Color")]
        public Color Color
        {
            get { return menmColor; }
            set { menmColor = value; }
        }

        /// <summary>
        /// String presentation of DemoObject
        /// </summary>
        public override string ToString()
        {
            return string.Format(
@"
StringValueA:{0}
StringValueB:{1}
StringValueC:{2}
ValueA:{3}
ValueB:{4}
ValueC:{5}
Color:{6}
EnumProperty:{7}
",
            this.StringValueA, this.StringValueB, this.StringValueC,
            this.ValueA, this.ValueB, this.ValueC,
            this.Color.ToString(),
            this.EnumProperty.ToString());
        }
    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/PropertyGrid/Functionality/DemoObject.vb
Imports System.Drawing
Imports System.ComponentModel

Namespace CompanionKit.Controls.PropertyGrid.Functionality

	''' <summary>
	''' The enum to use to demostrate editing of a property in the PropertyGrid
	''' </summary>
	Public Enum DemoEnum
		EnumValueA = 1
		EnumValueB = 2
		EnumValueC = 4
	End Enum

	''' <summary>
	''' A demo object to use to demostrate editing of a property 
	''' in the PropertyGrid.
	''' </summary>
    <TypeConverter(GetType(DemoObjectConverter))> _
	Public Class DemoObject


		' Fields
		Private menmColor As Color = Color.LawnGreen
		Private menmValueD As DemoEnum = DemoEnum.EnumValueA
		Private mintValueA As Integer = 1
		Private mintValueB As Integer = 2
		Private mintValueC As Integer = 3
		Private mstrValueA As String = "Initial A"
		Private mstrValueB As String = "Initial B"
		Private mstrValueC As String = "Initial C"

		' Properties
		<Description("The property with enum type - Color"), _
		   Category("Enums")> _
		Public Property Color() As Color
			Get
				Return Me.menmColor
			End Get
			Set(ByVal value As Color)
				Me.menmColor = value
			End Set
		End Property

		<Description("The property with enum type"), _
		Category("Enums")> _
		Public Property EnumProperty() As DemoEnum
			Get
				Return Me.menmValueD
			End Get
			Set(ByVal value As DemoEnum)
				Me.menmValueD = value
			End Set
		End Property

		<Category("Strings"), _
		Description("String - 'A' Read/Write property in Strings category")> _
		Public Property StringValueA() As String
			Get
				Return Me.mstrValueA
			End Get
			Set(ByVal value As String)
				Me.mstrValueA = value
			End Set
		End Property

		<Category("Strings"), _
		Description("String - 'B' Read/Write property in Strings category")> _
		Public Property StringValueB() As String
			Get
				Return Me.mstrValueB
			End Get
			Set(ByVal value As String)
				Me.mstrValueB = value
			End Set
		End Property

		<Category("Advanced"), _
		Description("String - 'C' Read/Write property in Advanced category")> _
		Public Property StringValueC() As String
			Get
				Return Me.mstrValueC
			End Get
			Set(ByVal value As String)
				Me.mstrValueC = value
			End Set
		End Property

		<Description("Integer ReadOnly property"), _
		Browsable(True), _
		[ReadOnly](True), _
		Category("Int Properties")> _
		Public Property ValueA() As Integer
			Get
				Return Me.mintValueA
			End Get
			Set(ByVal value As Integer)
				Me.mintValueA = value
			End Set
		End Property

		<Category("Int Properties"), _
		Description("Integer Read/Write property")> _
		Public Property ValueB() As Integer
			Get
				Return Me.mintValueB
			End Get
			Set(ByVal value As Integer)
				Me.mintValueB = value
			End Set
		End Property

		<Category("Advanced"), _
		Description("Integer Read/Write property in Advanced category")> _
		Public Property ValueC() As Integer
			Get
				Return Me.mintValueC
			End Get
			Set(ByVal value As Integer)
				Me.mintValueC = value
			End Set
		End Property


		''' <summary>
		''' String presentation of DemoObject
		''' </summary>
		Public Overrides Function ToString() As String
			Return String.Format(vbCrLf + "StringValueA:{0}" + vbCrLf + "StringValueB:{1}" + vbCrLf _
			  + "StringValueC:{2}" + vbCrLf + "ValueA:{3}" + vbCrLf + "ValueB:{4}" + vbCrLf _
			  + "ValueC:{5}" + vbCrLf + "Color:{6}" + vbCrLf + "EnumProperty:{7}" + vbCrLf, _
			  New Object() {Me.StringValueA, Me.StringValueB, Me.StringValueC, Me.ValueA, Me.ValueB, _
			  Me.ValueC, Me.Color.ToString, Me.EnumProperty.ToString})
		End Function
	End Class
End Namespace

